astropy/timeseries/periodograms/lombscargle/setup_package.py

```diff
@@ -1,18 +1,19 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 import os
+from pathlib import Path
 
 from numpy import get_include as get_numpy_include
 from setuptools import Extension
 
-ROOT = os.path.relpath(os.path.dirname(__file__))
+ROOT = os.path.relpath(Path(__file__).parent)
 
 
 def get_extensions():
     ext = Extension(
         "astropy.timeseries.periodograms.lombscargle.implementations.cython_impl",
         define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
-        sources=[os.path.join(ROOT, "implementations", "cython_impl.pyx")],
+        sources=[str(Path(ROOT) / "implementations" / "cython_impl.pyx")],
```

Comment:
Does Extension not work with Path objects?

Reply:
Nope, https://github.com/pypa/distutils/blob/e651e536ef1fbafbe48856815602127df0371835/distutils/extension.py#L109

I guess I can try getting this fixed upstream :)

This problem also pops up for `extension_helpers`, I'm prepping a PR to move to pathlib.